### PR TITLE
[HxInputTags] Missing Label if LabelType="LabelType.Floating" #166

### PR DESCRIPTION
**Pull request by Harvey1214**

### CODE REVIEW

File: Havit.Blazor.Components.Web.Bootstrap/Tags/Internal/HxInputTagsInternal.razor [hunk 1]
```diff
@@ -1,6 +1,6 @@
 @namespace Havit.Blazor.Components.Web.Bootstrap.Internal
 
-<span class="@CssClassHelper.Combine((HasInputGroupsEffective ? "input-group" : null), InputSizeEffective.AsInputGroupCssClass(), InputGroupCssClass)">
+<span class="@CssClassHelper.Combine((HasInputGroupsEffective ? "input-group" : null), InputSizeEffective.AsInputGroupCssClass(), InputGroupCssClass, "form-floating")">
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Tags/Internal/HxInputTagsInternal.razor:3]: I doubt we want the `form-floating` class to be there all the time....?

File: Havit.Blazor.Components.Web.Bootstrap/Tags/Internal/HxInputTagsInternal.razor.cs [hunk 2]
```diff
@@ -111,6 +111,11 @@ public partial class HxInputTagsInternal
 		/// </summary>
 		[Parameter] public RenderFragment InputGroupEndTemplate { get; set; }
 
+		/// <summary>
+		/// Label to render before input.
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Tags/Internal/HxInputTagsInternal.razor.cs:115]: The `Label` here is not rendered before the input, as far as I see. Should we say "Label (only rendered for LabelType.Floating. The wrapping HxInputTags renders the label on his own for other cases)"? Or is it used in other scenarios?

File: Havit.Blazor.Components.Web.Bootstrap/Tags/Internal/HxInputTagsInternal.razor.cs [hunk 3]
```diff
@@ -122,6 +127,7 @@ public partial class HxInputTagsInternal
 		protected bool HasInputGroupsEffective => !String.IsNullOrWhiteSpace(InputGroupStartText) || !String.IsNullOrWhiteSpace(InputGroupEndText) || (InputGroupStartTemplate is not null) || (InputGroupEndTemplate is not null);
 
 		private string dropdownId = "hx" + Guid.NewGuid().ToString("N");
+		private string InputTagsControlId = "hx" + Guid.NewGuid().ToString("N");
```
Comment (hakenr) [inline Havit.Blazor.Components.Web.Bootstrap/Tags/Internal/HxInputTagsInternal.razor.cs:130]: If it is private, it has to be lowerCase.

**Review by hakenr — CHANGES_REQUESTED** [review]

File: Havit.Blazor.Components.Web.Bootstrap/Tags/Internal/HxInputTagsInternal.razor (hunk 1 above)
Comment (Harvey1214) [reply Havit.Blazor.Components.Web.Bootstrap/Tags/Internal/HxInputTagsInternal.razor:3]: @hakenr I agree. This is still a work-in-progress - the label shows correctly but doesn't animate and fill the input when there is no value entered.

**Comment by Harvey1214** [comment]
@crdo I've managed to get the floating label working when the HxInputTags component has some values, however, if the component is empty, the label stays in position and doesn't animate due to the current structure of the component.

I've thought of a solution but it's not ideal and the animation either wouldn't work or we would have to remake it.

I'd gladly appreciate any ideas for solutions to this issue, thanks :)

**Comment by crdo** [comment]
@Harvey1214 I played around with it for a while and came to a conclusion that with the current design of floating labels in Bootstrap and HxInputTags in Havit.Blazor it would be very complicated to make this working.

**Pull request closed without merging**